From 658c034d92027dc8af5f784cae852123fac79b19 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 16 Apr 2016 13:28:59 -0700
Subject: [PATCH] Do not ignore return value of write()

gcc warns about ignoring return value when compiling
with fortify turned on.

assert when write() fails

Upstream-Status: Submitted
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 glib/tests/unix.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/glib/tests/unix.c b/glib/tests/unix.c
index 7639d06..f941141 100644
--- a/glib/tests/unix.c
+++ b/glib/tests/unix.c
@@ -33,14 +33,15 @@ test_pipe (void)
   GError *error = NULL;
   int pipefd[2];
   char buf[1024];
-  gssize bytes_read;
+  gssize bytes_read, bytes_written;
   gboolean res;
 
   res = g_unix_open_pipe (pipefd, FD_CLOEXEC, &error);
   g_assert (res);
   g_assert_no_error (error);
 
-  write (pipefd[1], "hello", sizeof ("hello"));
+  bytes_written = write (pipefd[1], "hello", sizeof ("hello"));
+  g_assert (bytes_written != -1 && "write() failed");
   memset (buf, 0, sizeof (buf));
   bytes_read = read (pipefd[0], buf, sizeof(buf) - 1);
   g_assert_cmpint (bytes_read, >, 0);
